bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCheckIssueVisitor.java

```diff
@@ -54,16 +54,27 @@ class PullRequestCheckIssueVisitor implements IssueVisitor {
         readyForReview = true;
     }
 
-    private void addFailureMessage(Check check, String message) {
-        failedChecks.put(check.getClass(), message);
+    private void addMessage(Check check, String message, Severity severity, boolean readyForReviewWhenFailedAsError) {
+        if (severity.equals(Severity.ERROR)) {
+            errorFailedChecks.put(check.getClass(), message);
+            if (this.readyForReview) {
+                this.readyForReview = readyForReviewWhenFailedAsError;
+            }
```

Comment:
This part is rather confusing to try to understand. Maybe we need a separate method for updating the `readyForReview` field, and remove the new parameter `readyForReviewWhenFailedAsError` from this method, to make this interaction clearer? 

```
private void setNotReadyForReviewOnError(Severity severity) {
    if (severity == ERROR) {
        readyForReview = false;
    }
}
```

This method would be called from the places that used to do `readyForReview = false`.